### PR TITLE
Bp 392 eventdiagram

### PR DESCRIPTION
**Pull request by MaximilianV**
Adds the known event diagram to the detail page.
This version will also update with applied filters.

### CODE REVIEW

File: src/DetailView/EventDiagram.js [hunk 1]
```diff
@@ -1,12 +1,104 @@
...
+			},
+			xaxis: {
+				showgrid: false,
+			},
+			height: 400
+		};
+
+		const statusColor = Utils.getStatus(entity.Status).color;
+
+		let diagramData = {
```
Comment (hpibptlab) [inline src/DetailView/EventDiagram.js]: MINOR Make "diagramData" "const". [rule](http://bpt-lab.org/sonarqube/coding_rules#rule_key=javascript%3AS3353)

**Comment by hpibptlab** [comment]
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

File: src/DetailView/EventDiagram.js [hunk 2]
```diff
@@ -1,12 +1,104 @@
...
 class EventDiagram extends Component {
+
+	static sortEventsByTime(eventA, eventB, timeStampAttributeName) {
+		const dateA = new Date(EventDiagram.getTimeStampAsStringFromEvent(eventA, timeStampAttributeName));
+		const dateB = new Date(EventDiagram.getTimeStampAsStringFromEvent(eventB, timeStampAttributeName));
+		return dateA - dateB;
+	}
+
+	static getTimeStampAsStringFromEvent(event, timeStampAttributeName) {
+		return event.Attributes.find(attribute =>
```
Comment (friedow) [inline src/DetailView/EventDiagram.js:20]: Is ensured, that `find` will find something? If not you'll try to receive `.Value` of undefined.

File: src/DetailView/EventDiagram.js [hunk 3]
```diff
@@ -1,12 +1,104 @@
...
+		return EventDiagram
+			.getTimeStampAsStringFromEvent(event, timeStampAttributeName)
+			.split("T")[0];
+	}
+
+	componentWillReceiveProps(props) {
+		this.buildChartDataset(props);
+	}
+
+	buildChartDataset(props) {
```
Comment (friedow) [inline src/DetailView/EventDiagram.js:70]: Please split this function. It's way too long.

File: src/DetailView/EventDiagram.js [hunk 4]
```diff
@@ -1,12 +1,104 @@
...
+		const {events, eventType, eventTypeAttributes, entity} = props;
+		if (!events) {
+			return;
+		}
+
+		const x = [];
+		const y = [];
+		const dateCounter = {};
+		let eventCounter = 0;
+		const timeStampAttributeName = eventTypeAttributes.find(attribute =>
```
Comment (friedow) [inline src/DetailView/EventDiagram.js:80]: Is ensured, that `find` will find something? If not you'll try to receive `.Name` of undefined.

File: src/DetailView/EventDiagram.js [hunk 5]
```diff
@@ -1,12 +1,104 @@
...
+			y: y,
+			type: 'line',
+			line: {
+				color: statusColor
+			}
+		};
+
+		if (eventCounter !== 1) {
+			diagramData.fill = 'tozeroy';
+			diagramData.fillcolor = statusColor + '80';
```
Comment (friedow) [inline src/DetailView/EventDiagram.js]: What does `+80` do ?

File: src/DetailView/EventTabs.js [hunk 6]
```diff
@@ -21,7 +20,7 @@ class EventTabs extends Component {
 					return (
 						<Tab
 							label={eventType.Name}
-							onClick={() => this.props.eventTypeChangeHandler((eventType.Id))}
+							onClick={() => this.props.onEventTypeChange((eventType))}
```
Comment (friedow) [inline src/DetailView/EventTabs.js:23]: Please investigate wether to use `onClick` or `onTouchTap`.

**Review by friedow — CHANGES_REQUESTED** [review]

File: src/DetailView/EventDiagram.js (hunk 2 above)
Comment (MaximilianV) [reply src/DetailView/EventDiagram.js:20]: If we assume, that all events are complete, there should always be an timestamp attribute :) @Hannes01071995 confirmed

File: src/DetailView/EventDiagram.js (hunk 4 above)
Comment (MaximilianV) [reply src/DetailView/EventDiagram.js:80]: Every eventtype has a timestamp attribute

File: src/DetailView/EventDiagram.js (hunk 5 above)
Comment (MaximilianV) [reply src/DetailView/EventDiagram.js]: it adds transparency

File: src/DetailView/EventTabs.js (hunk 6 above)
Comment (MaximilianV) [reply src/DetailView/EventTabs.js:23]: doesn't work

File: src/DetailView/EventDiagram.js [hunk 7]
```diff
@@ -1,12 +1,117 @@
...
+				x.push(eventDate);
+			}
+			else {
+				dateCounter[eventDate] = currentDateCount + 1;
+			}
+		});
+
+		//build y-axis
+		x.forEach(date =>
+			y.push(dateCounter[date]));
```
Comment (hpibptlab) [inline src/DetailView/EventDiagram.js]: MAJOR Remove this logging statement. [rule](http://bpt-lab.org/sonarqube/coding_rules#rule_key=javascript%3AS2228)

**Comment by hpibptlab** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

**Review by friedow — APPROVED** [review]

**Pull request merged**